support/cas-server-support-cassandra-ticket-registry/src/main/java/org/apereo/cas/dao/CassandraDao.java

```diff
@@ -0,0 +1,242 @@
+                        final TicketSerializer<T> serializer, final Class<T> typeToWriteToCassandra) {
+        this.serializer = serializer;
+        this.typeToWriteToCassandra = typeToWriteToCassandra;
+        final Cluster cluster = Cluster.builder().addContactPoints(contactPoints.split(",")).withCredentials(username, password)
+                .withProtocolVersion(ProtocolVersion.V3).build();
+
+        this.session = cluster.connect(KEYSPACE);
+
+        this.selectTgtStmt = session.prepare(SELECT_TGT).setConsistencyLevel(ConsistencyLevel.LOCAL_QUORUM);
+        this.insertTgtStmt = session.prepare(INSERT_TGT).setConsistencyLevel(ConsistencyLevel.LOCAL_QUORUM);
+        this.deleteTgtStmt = session.prepare(DELETE_TGT).setConsistencyLevel(ConsistencyLevel.LOCAL_QUORUM);
+        this.updateTgtStmt = session.prepare(UPDATE_TGT).setConsistencyLevel(ConsistencyLevel.LOCAL_QUORUM);
+
+        this.selectStStmt = session.prepare(SELECT_ST).setConsistencyLevel(ConsistencyLevel.LOCAL_ONE);
+        this.insertStStmt = session.prepare(INSERT_ST).setConsistencyLevel(ConsistencyLevel.LOCAL_ONE);
+        this.deleteStStmt = session.prepare(DELETE_ST).setConsistencyLevel(ConsistencyLevel.LOCAL_ONE);
+        this.updateStStmt = session.prepare(UPDATE_ST).setConsistencyLevel(ConsistencyLevel.LOCAL_ONE);
+
+        this.insertExStmt = session.prepare(INSERT_EX).setConsistencyLevel(ConsistencyLevel.LOCAL_QUORUM);
+        this.deleteExStmt = session.prepare(DELETE_EX).setConsistencyLevel(ConsistencyLevel.LOCAL_QUORUM);
+        this.selectExStmt = session.prepare(SELECT_EX).setConsistencyLevel(ConsistencyLevel.LOCAL_QUORUM);
+        this.selectDateExStmt = session.prepare(SELECT_DATE_EX).setConsistencyLevel(ConsistencyLevel.LOCAL_QUORUM);
+
+        this.selectLrStmt = session.prepare(SELECT_LR).setConsistencyLevel(ConsistencyLevel.LOCAL_QUORUM);
+        this.updateLrStmt = session.prepare(UPDATE_LR).setConsistencyLevel(ConsistencyLevel.LOCAL_QUORUM);
+
+        final long lastRun = getLastRunTimestamp();
+        final long currentTime = currentTimeBucket();
+        if (lastRun == 0 || lastRun > currentTime) {
+            updateLastRunTimestamp(currentTime);
+        }
+    }
+
+    @Override
+    public void addTicketGrantingTicket(final Ticket ticket) {
+        LOGGER.debug("INSERTING TICKET {}", ticket.getId());
+        session.execute(this.insertTgtStmt.bind(ticket.getId(), serializer.serializeTGT(ticket)));
+        final TicketGrantingTicketImpl tgt = (TicketGrantingTicketImpl) ticket;
+        addTicketToExpiryBucket(ticket, calculateExpirationDate(tgt));
+    }
+
+    @Override
+    public void addServiceTicket(final Ticket ticket) {
+        LOGGER.debug("INSERTING TICKET {}", ticket.getId());
+        session.execute(this.insertStStmt.bind(ticket.getId(), serializer.serializeST(ticket)));
+    }
+
+    @Override
+    public boolean deleteTicketGrantingTicket(final String id) {
+        LOGGER.debug("DELETING TICKET {}", id);
```

Comment:
Correct all log messages and remove caps. This should be "Deleting ticket {}", etc.